### ISSUE TITLE
False negative with timeout

### ISSUE THREAD

**Issue by falsandtru** [makenowjust-labs/recheck#870]
# What is the problem?

Setting `recallTimeout` or `seedingTimeout` to null hides errors. That setting just enables unlimited checking, right?

https://github.com/falsandtru/securemark/blob/v0.274.2/.eslintrc.json

`/a+$/`: Found a ReDoS vulnerable RegExp (2nd degree polynomial)

# How to fix the problem?

Remove those settings.

# Related issues

**Comment by makenowjust** [comment]
There are some problems and notes.

- `seedingTimeout` is unrelated. At least, `recheck` does not refer to this value in your example `/a*$/`.
- `recallTimeout: null` does not make sense. Because the ReDoS vulnerability is reported when the match time exceeds this threshold, so `recallTimeout: null` will report that there is no ReDoS vulnerability, no matter how long it takes to match.
- Finally, the recall validation does not work in the `pure` implementation of Node.js. We should not set `recallTimeout`.
  * I expected the latest version (`4.5.0-beta.3`) supports this value. However, it does not work due to the BUG. I would fix this BUG.

Essentially, I don't think there is an issue, but I would like to keep this issue open until the above-mentioned bug is fixed.

**Comment by falsandtru** [comment]
Still not fixed.

Without `recallTimeout: null`:
https://github.com/falsandtru/spica/actions/runs/4848224243/jobs/8639151089#step:5:297

With `recallTimeout: null`:
https://github.com/falsandtru/spica/actions/runs/4848226107/jobs/8639154646

**Comment by makenowjust** [comment]
`/a+$/` is vulnerable. It is just a **TRUE** positive.

**Comment by falsandtru** [comment]
I mean `recallTimeout: null` hides that true positive.

**Comment by makenowjust** [comment]
It is expected behavior. Recall,

> - recallTimeout: null does not make sense. Because the ReDoS vulnerability is reported when the match time exceeds this threshold, so recallTimeout: null will report that there is no ReDoS vulnerability, no matter how long it takes to match.

**Comment by makenowjust** [comment]
If `recallTimeout` is `null`, it may be an error or a warning.

**Comment by falsandtru** [comment]
I was not sure about that behavior, thanks.

**Comment by falsandtru** [comment]
However, in the document, it is unclear that the difference of behaviors with null between timeout parameters. The document notes null only with `timeout` parameter. The other timeout parameters such as `seedingTimeout` and `recallTimeout` gives no hints of null. Nobody could read their inconsistent behaviors with null.

**Comment by makenowjust** [comment]
Behaviors on `timeout` and `recallTimeout` are consistent; `null` means the infinite duration. However, I have used `timeout` in two contexts: One is a timeout for `recheck` (step) execution, and the other is a timeout for a matching time to determine ReDoS or not. I will find better wording for them.

Most of the parameters were added for my own experimentation and are not intended to be set by users. I have no time to enhance the document. Sorry.
